runtime/src/main/java/org/corfudb/runtime/CheckpointLivenessUpdater.java

```diff
@@ -52,24 +58,35 @@ private Table<TableName, ActiveCPStreamMsg, Message> getActiveCheckpointsTable()
                 TableOptions.fromProtoSchema(ActiveCPStreamMsg.class));
     }
 
-    private void addTask() {
+    private void updateHeartbeat() {
         if (!currentTable.isPresent()) {
             return;
         }
-        executorService.execute(() -> {
-            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-                ActiveCPStreamMsg currentStatus = (ActiveCPStreamMsg)
-                        txn.getRecord(CompactorMetadataTables.ACTIVE_CHECKPOINTS_TABLE_NAME, currentTable.get()).getPayload();
-                ActiveCPStreamMsg newStatus = ActiveCPStreamMsg.newBuilder()
-                        .setSyncHeartbeat(currentStatus.getSyncHeartbeat() + 1)
-                        .build();
-                // update validity counter for the current table
-                txn.putRecord(activeCheckpointsTable, currentTable.get(), newStatus, null);
-                txn.commit();
-            } catch (Exception e) {
-                log.error("Unable to update liveness for table: {} due to exception: {}", currentTable, e.getStackTrace());
+
+        TableName table;
+        try {
+            table = currentTable.get();
+        } catch (NoSuchElementException e) {
+            log.info("Encountered NoSuchElementException while accessing currentTable, ", e);
+            return;
+        }
+
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            ActiveCPStreamMsg currentStatus = (ActiveCPStreamMsg)
+                    txn.getRecord(CompactorMetadataTables.ACTIVE_CHECKPOINTS_TABLE_NAME, table).getPayload();
+            ActiveCPStreamMsg newStatus = ActiveCPStreamMsg.newBuilder()
+                    .setSyncHeartbeat(currentStatus.getSyncHeartbeat() + 1)
+                    .build();
+            // update validity counter for the current table
+            txn.putRecord(activeCheckpointsTable, table, newStatus, null);
+            txn.commit();
+        } catch (TransactionAbortedException ex) {
+            if (ex.getAbortCause() == AbortCause.CONFLICT) {
+                log.warn("Another node tried to commit, ", ex);
```

Comment:
This message seems misleading - only the leader node can update the heartbeat. When it hit CONFLICT here, the only case I can think of is the unlockTableAfterCheckpoint deletes the entry.

Reply:
Probably also include the `table` in the log message to help debug.
Failed to increment the heartbeat for table xxx due to transaction conflict.